### PR TITLE
doc: Build static documentation with semiliterate

### PR DESCRIPTION
**Pull request by gwhitney**
This is the first step toward an integrated static documentation site. I am not proposing this be merged at this point. Rather, I am submitting it so that (a) I can make sure that the basic setup is workable for others, and (b) obtain some feedback as to the organization of the site. So if anyone viewing this repository has a few minutes to lend a hand, please pull this PR and try out/respond to the next couple of comments here.

### CODE REVIEW

**Comment by liam-mulhall** [comment]
I'm ready to be semiliterate! 😃

**Comment by gwhitney** [comment]
This first commit simply does the bare install of semiliterate and gets it to run with zero tailoring: default skin (not even "ReadTheDocs"), no new documentation, no extraction of documentation from source, no nuthin'. (I will add those things in future commits.) Neverthelesss, out of the box it produces a decently plausible documentation site, as I hope you will see if you try the following steps:
- check out/pull the `semiliterate` branch from my clone and make sure you are in the top-level directory of the project
- Run `npm install` -- this should produce a python virtual environment ".venv" in the top-level directory with semiliterate installed
- Activate that virtual environment in your favorite way (e.g., if you use bash, execute `source .venv/bin/activate`)
- Run `mkdocs serve`. This should start up a mini web server serving the documentation; the last thing it will do is print out a link
- Open that link in your favorite browser, and leaf around in the resulting documentation site

Please can anyone/everyone let me know if this works for them? It all goes smoothly for me but I don't have a lot of experience with other setups, so I want to make sure that the basic operation is sound. Especially if you run into problems, please let me know as soon as convenient. (The next comment has some specific questions about the resulting site.)

**Comment by liam-mulhall** [comment]
The instructions in https://github.com/numberscope/frontscope/pull/139#issuecomment-1182471683 worked perfectly for me.

**Comment by gwhitney** [comment]
Specific questions:
1) Do we want the separate LICENSE page on our doc site, or is the LICENSE link at the top of the README file (which will also serve as the overview/table of contents for the doc site as a whole) sufficient?
2) The scope and organization of the README is pretty appropriate for the "front page" of a GitHub repository, but it's a bit off kilter for the beginning of a general documentation site. For example, the beginning of a "user guide" or at least a link to it should be much more prominent, whereas details about running from source code would be relegated to other detail pages (since most casual visitors would not be interested in that). I see two possible solutions:
    a) Trim the README down to the "least common denominator" between these two purposes: just the top-level overview and orientation to the reader, basically followed by a list of topics and links in order from most general purpose (user guide for a casual visitor) to most detailed/specific (e.g. the nitty gritty of submitting a PR, say).
    b) Basically leave the README as it is for the repository, but use the extraction power of semiliterate to slice it into two or more pieces for presentation on the site. This will involve adding some comments to it (that will not be rendered by GitHub) so that semiliterate knows where to slice it. I have done this sort of thing before (basically if you look carefully at the README at the bottom of https://code.studioinfinity.org/glen/mkdocs-semiliterate (the equivalent of the GitHub link) versus the front page of the official documentation at http://studioinfinity.org/semiliterate you will see they are not quite identical. This approach is just a bit more complex and I'd be happy to avoid it if there is a viable nugget of the README that will basically work in both places.

Let me know your thoughts on these questions, and of course I also welcome any other particular suggestions, concerns, or thoughts about the developing doc site. Thanks!

**Comment by gwhitney** [comment]
> The instructions in [#139 (comment)](https://github.com/numberscope/frontscope/pull/139#issuecomment-1182471683) worked perfectly for me.

That is a big relief to hear. @katestange ?

**Comment by liam-mulhall** [comment]
> Do we want the separate LICENSE page on our doc site, or is the LICENSE link at the top of the README file (which will also serve as the overview/table of contents for the doc site as a whole) sufficient?

I think a link at the top of the README is sufficient.

> The scope and organization of the README is pretty appropriate for the "front page" of a GitHub repository, but it's a bit off kilter for the beginning of a general documentation site.

Agreed. I prefer solution (a).

**Comment by gwhitney** [comment]
OK, so one potentially key point about trying to use a stripped down README in both places is simply that top-line header. For GitHub, it reads "frontscope: Numberscope's Front End" which seems entirely appropriate. On the other hand, I would think the general doc site would just be titled "Numberscope" and any aspects relating to the division between front and back scopes would be relegated to a deep detail page somewhere.

So, is there a common heading that works in both circumstances? I am definitely hoping for some suggestions on this point, which on the one hand seems sort of trivial, but on the other hand, that heading is the first thing anyone sees...

**Comment by gwhitney** [comment]
Yes, if you look at the markdown source for mkdocs-semiliterate/README.md, you will see exactly that it has HTML comments to provide alternate versions of some content for the "repo" and the "site". That sort of thing was roughly what I was trying to avoid via option (a) rather than (b). But if it's the best way to go, we definitely can put structured comments in the README to control what's visible in each context.

**Comment by katestange** [comment]
When I run ```npm install```, I get this:
```
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!

> yorkie@2.0.0 uninstall /home/katestange/data/projects/numberscope/frontscope-backscope/frontscope/node_modules/yorkie
> node bin/uninstall.js

husky
uninstalling Git hooks
done


> core-js@3.23.3 postinstall /home/katestange/data/projects/numberscope/frontscope-backscope/frontscope/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js:
> https://opencollective.com/core-js 
> https://patreon.com/zloirock 
> bitcoin: bc1qlea7544qtsmj2rayg0lthvza9fau63ux0fstcz 

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)


> esbuild@0.14.23 postinstall /home/katestange/data/projects/numberscope/frontscope-backscope/frontscope/node_modules/esbuild
> node install.js


> frontscope@0.3.0 postinstall /home/katestange/data/projects/numberscope/frontscope-backscope/frontscope
> python3 -m venv .venv && .venv/bin/pip install -U pip && .venv/bin/pip install -r requirements.txt

The virtual environment was not created successfully because ensurepip is not
available.  On Debian/Ubuntu systems, you need to install the python3-venv
package using the following command.

    apt install python3.8-venv

You may need to use sudo with that command.  After installing the python3-venv
package, recreate your virtual environment.

Failing command: ['/home/katestange/data/projects/numberscope/frontscope-backscope/frontscope/.venv/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! frontscope@0.3.0 postinstall: `python3 -m venv .venv && .venv/bin/pip install -U pip && .venv/bin/pip install -r requirements.txt`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the frontscope@0.3.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/katestange/.npm/_logs/2022-07-13T16_06_18_541Z-debug.log

```

Installing the missing package fixes the problem.  Do any of these warnings matter?
```
npm WARN frontscope@0.3.0 No repository field.
npm WARN frontscope@0.3.0 No license field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-darwin-64@0.14.23 (node_modules/esbuild-darwin-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-darwin-64@0.14.23: wanted {"os":"darwin","arch":"x64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-android-arm64@0.14.23 (node_modules/esbuild-android-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-android-arm64@0.14.23: wanted {"os":"android","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-freebsd-64@0.14.23 (node_modules/esbuild-freebsd-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-freebsd-64@0.14.23: wanted {"os":"freebsd","arch":"x64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-darwin-arm64@0.14.23 (node_modules/esbuild-darwin-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-darwin-arm64@0.14.23: wanted {"os":"darwin","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-freebsd-arm64@0.14.23 (node_modules/esbuild-freebsd-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-freebsd-arm64@0.14.23: wanted {"os":"freebsd","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-arm@0.14.23 (node_modules/esbuild-linux-arm):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-arm@0.14.23: wanted {"os":"linux","arch":"arm"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-mips64le@0.14.23 (node_modules/esbuild-linux-mips64le):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-mips64le@0.14.23: wanted {"os":"linux","arch":"mips64el"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-arm64@0.14.23 (node_modules/esbuild-linux-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-arm64@0.14.23: wanted {"os":"linux","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-32@0.14.23 (node_modules/esbuild-linux-32):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-32@0.14.23: wanted {"os":"linux","arch":"ia32"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-ppc64le@0.14.23 (node_modules/esbuild-linux-ppc64le):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-ppc64le@0.14.23: wanted {"os":"linux","arch":"ppc64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-riscv64@0.14.23 (node_modules/esbuild-linux-riscv64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-riscv64@0.14.23: wanted {"os":"linux","arch":"riscv64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-s390x@0.14.23 (node_modules/esbuild-linux-s390x):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-s390x@0.14.23: wanted {"os":"linux","arch":"s390x"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-netbsd-64@0.14.23 (node_modules/esbuild-netbsd-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-netbsd-64@0.14.23: wanted {"os":"netbsd","arch":"x64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-sunos-64@0.14.23 (node_modules/esbuild-sunos-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-sunos-64@0.14.23: wanted {"os":"sunos","arch":"x64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-openbsd-64@0.14.23 (node_modules/esbuild-openbsd-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-openbsd-64@0.14.23: wanted {"os":"openbsd","arch":"x64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-windows-32@0.14.23 (node_modules/esbuild-windows-32):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-windows-32@0.14.23: wanted {"os":"win32","arch":"ia32"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-windows-64@0.14.23 (node_modules/esbuild-windows-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-windows-64@0.14.23: wanted {"os":"win32","arch":"x64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-windows-arm64@0.14.23 (node_modules/esbuild-windows-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-windows-arm64@0.14.23: wanted {"os":"win32","arch":"arm64"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
```

The rest of the instructions run smoothly and I am now going to look around in the resulting site.

**Comment by liam-mulhall** [comment]
> When I run npm install, I get this:

```
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
```

This makes me wonder if you have an outdated version of Node. What's the output of `node -v`?

> Do any of these warnings matter?

I'm not sure. If the site seems to be working, I wouldn't worry about them too much. It's strange you're getting those warnings though.

**Comment by katestange** [comment]
For your specific questions:
1.  The MIT license seems to want you to include the text in your project, so having a license page is in that spirit.  But on the other hand, if it's included in the github codebase, maybe that's overkill.  We were perfectly happy to just have a link on the website when we were updating vue and looking at the homepage.  So I'm happy either way.
2. I like option (a) and the solution (using Github's About) that Glen and Liam worked out.

**Comment by katestange** [comment]
```node -v``` gives ```v10.19.0```

**Comment by liam-mulhall** [comment]
Ah, yeah that's outdated. The newest LTS is 16.16.0. I would try updating Node: https://nodejs.org/en/download/package-manager/#debian-and-ubuntu-based-linux-distributions.

**Comment by katestange** [comment]
> OK, I have updated the install instructions based on the error you got, and I agree that the warnings are mostly are all due to being several versions behind current on Node.

Yes, they go away with the new node.

**Comment by gwhitney** [comment]
OK, this is a "progress update" commit. I've tried to tailor the README to be suitable both for the docsite and github. You can see how it looks for the docsite if you pull the commit and do `mkdocs serve`, and you can see how it looks on GitHub at  https://github.com/gwhitney/frontscope/tree/semiliterate. On the docsite I have switched to the "ReadTheDocs" theme so you can see how that looks. I really like how its outline works much better, even though I think the default theme is more attractive. So I'd recommend sticking with this perhaps just with the color palette harmonized with Numberscope and a mock topbar like Numberscope's inserted.

Note there is still leftover material in the README from its previous incarnation that hasn't yet found its proper home in the new organization (it's labeled as such).

In this commit I've tried to spec out the basic structure of the docsite that I have in mind. There are also the first blobs of documentation that are extracted from src/views/Scope.vue and src/visualizers/VisualizerDifferences.ts so by looking at the changed files you can see how that will look.

If you have a chance to pull this and poke around I am happy for any feedback esp. on the suitability of the general direction here, but also on specific matters if you like. But it's not holding me up, I can continue working on the conversion, with (or without) feedback whenever it comes in. Thanks!

**Comment by gwhitney** [comment]
[Oops I forgot an added plugin in requirements.txt the first time, but it should be ok to pull now -- just remember to run `npm install` again after you do.]

**Comment by liam-mulhall** [comment]
I pulled the most recent changes, and I think it's looking good, @gwhitney. I don't have any advice. Thanks for doing this!

**Comment by gwhitney** [comment]
OK, I can definitely use additional feedback at this point. The latest commit completes my proposed changes to the content/organization of the documentation. There are additional documentation issues open, but I think they should correspond to a further PR or PRs. (This PR is nevertheless not ready for merging because I still need to enhance the `build` script to produce the docsite in a way that's available for preview; remove the About and Acknowledgment views and replace them with links to the docsite; and theme/skin the docsite as preferred.) So:
1. If you have any suggestions/critiques of specific aspects of the revised documentation content/organization, now is the time.
2. I presume I should add a package script that previews the documentation, along the lines of the 'dev' script that puts up the numberscope tool itself. I am at a bit of a loss of what to call it: 'docdev' or 'dev-docs'? Just 'docs'? (My only reluctance with the latter is that it might sound like it builds the docs, but my plan is to just fold building the docs into the 'build' script, so this script is just for showing the docs in a "live" fashion like 'dev' does. But it's not really practical to merge the docs one with 'dev' itself, because then one command would have to run two servers at once, and I think it's unlikely that you'll really be developing the docs and the tool _simultaneously_.) Other suggestions for the name of this script? Thanks.
3. For theming/skinning, I understand the plan is to add a top bar visually identical to the one in the Numberscope tool, but below that, what's the preferred them? This is showing the ReadTheDocs theme, which in my opinion is very functional but a bit drab. The first pass had the default MkDocs theme, which I think looks punchier but doesn't navigate as nicely. You can switch back to that first option to compare by changing 'readthedocs' to 'mkdocs' in the 'mkdocs.yml' file and re-running 'mkdocs serve'. And you can see the community-contributed themes https://github.com/mkdocs/mkdocs/wiki/MkDocs-Themes -- if any look good, it's easy to install and try out. Note that whatever of the existing themes we choose, we will be deriving from it to either add a top bar identical to the frontscope tool, or to modify the existing top bar of the theme to mostly match the frontscope tool.
4. (Deferred to its own comment below because it's long.)

**Comment by gwhitney** [comment]
This is item 4 from the list in the previous comment. The issue here is the documentation of the package scripts in package.json run with `npm run <SCRIPT>`. I strongly agree with Liam that they need to be documented, as we have some non-standard scripts and collectively they are an important part of our workflow. As you know, I also feel strongly that documentation should be co-located with implementation. In this case that runs up against the difficulty that npm package.json files are not allowed to have comments, and the npm maintenance group has repeatedly denied proposals that would allow comments.

As a result, I have adopted one of the various hacks to make the best of the situation, which you can see if you look at the scripts section of the package.json file. Basically, the idea is to include lots of extra script lines which both act as comments, and allow you to display those comments by running an appropriate npm script, e.g. `npm run help:lint` tells you what `npm run lint` does. Then I can extract from those extra script lines in the doc file on working with the package manager to incorporate documentation there. This does produce a workable situation, with at least two drawbacks: (i) the package.json file looks kinda crazy, and (ii) you will note there are no links on the working with the package manager page, whereas there were in the corresponding
section of the README file that all of that information came from. That's because there's basically no reasonable way to include
parentheses in these pseudo-comments, because parens are special characters in `sh` which runs the script lines.

So we could leave things this way (call that option (A)), but there are two alternatives that it's worth thinking about adopting instead. (I didn't bring up this consideration in advance because I didn't really think it would make sense until you could see the situation in package.json):

(B) Switch this project from using the `npm` package manager to exclusively using the `pnpm` package manager. (Unlike `npm` and `yarn`, the `pnpm` is not inter-operable with the others.) The advantage of this change is that `pnpm` supports the JSON5 standard for package files, and JSON5 explicitly allows ordinary comments, eliminating the need for "dummy" scripts and fancy extraction patterns -- the default extraction method of semiliterate can be used to pull Markdown out of the resulting package.json5 file. In addition, `pnpm` is benchmarked as considerably faster than `npm` (it stands for 'performant npm'), especially if you have multiple projects that use it on the same hard drive. On the other hand, `pnpm` is less common than `npm`; but the syntax for using it is almost identical, so it should be relatively easy for us and for folks familiar with `npm` to use. But it would be something that would be a bit "different" about this project than most, so if there's a desire to make everything here as "customary" as possible, then I could see why you might not want to switch to `pnpm`.

(C) If we don't like package.json the way it is in this commit, and also don't want to switch to `pnpm`: Because we currently only have documentation in the "scripts" section of package.json, there is another option, which is to install the `nps` package (https://www.npmjs.com/package/nps) with which one can move (just) the scripts section out of package.json and into an ordinary javascript file (conventionally called "package-scripts.js"). The drawbacks of this option are that it creates yet another top-level configuration file, and for full operation, it requires that users install the "nps" package globally, rather than just in this project. But again, as long as we only want documentation in the scripts section, option (C) will allow default extraction of Markdown from our files.

Any comments/thoughts/votes on whether we should choose option (A), (B), or (C) would be very welcome. 

(I will spill my beans and say that if it were entirely up to me, I would go with option (B) because I would be very happy for this project to vote with its feet and switch away from npm because of their dogmatic view that there is "no solution" to the difficulties of package.json, which seems shortsighted because (a) `pnpm` manages to support multiple formats, and (b) clearly eventually the world will move on in any case, no "standard" lasts forever. In particular, I think that enough projects switching to `pnpm` or other alternate package managers will put pressure on `npm` to improve. But I don't want to hold this project hostage to my coding-related semi-political views, hence the desire to get at least both @katestange and @liam-mulhall to weigh in on this.)

**Comment by liam-mulhall** [comment]
## General Feedback

* Looks very professional! I'm excited about this PR.
* I noticed the following error:
```
(.venv) liam@ratbook frontscope % mkdocs serve
INFO     -  Building documentation...
INFO     -  mkdocs-simple-plugin: build_docs_dir:
            /var/folders/6m/_m8brjsn2j1g4cp6sms16vqm0000gn/T/mkdocs_simple_frontscopebqy58fzl
ERROR    -  mkdocs-simple-plugin: could not build ./doc/working-with-pm.md
INFO     -  Cleaning site directory
INFO     -  Documentation built in 0.15 seconds
INFO     -  [14:06:36] Watching paths for changes:
            '/var/folders/6m/_m8brjsn2j1g4cp6sms16vqm0000gn/T/mkdocs_simple_frontscopebqy58fzl', 'mkdocs.yml', 'doc',
            'README.md', 'dist/assets/turtle.be13824c.png', 'dist/assets/sc.a8f491cd.png',
            'dist/assets/gof.ff388071.png', 'dist/assets/LogoWithMicroscope.1707fce7.png', 'doc/user_guide.md',
            'doc/running-from-source.md', 'doc/extending.md', 'doc/making-a-visualizer.md', 'doc/contributing.md',
            'doc/acknowledgments.md', 'doc/about.md', 'doc/visual-studio-code-setup.md',
            'src/visualizers/VisualizerTurtle.ts', 'src/visualizers/VisualizerShiftCompare.ts',
            'src/visualizers/VisualizerModFill.ts', 'src/visualizers/VisualizerChaos.ts',
            'src/visualizers/VisualizerDifferences.ts', 'src/assets/img/LogoWithMicroscope.png',
            'src/assets/img/LogoNoMicroscope.png', 'src/assets/img/VertLogoWithMicroscope.png',
            'src/assets/img/specimens/4.png', 'src/assets/img/specimens/5.png', 'src/assets/img/specimens/7.png',
            'src/assets/img/specimens/6.png', 'src/assets/img/specimens/2.png', 'src/assets/img/specimens/3.png',
            'src/assets/img/specimens/1.png', 'src/assets/img/homepage/turtle.png', 'src/assets/img/homepage/gof.png',
            'src/assets/img/homepage/sc.png', 'src/assets/img/toolpage/1-sequence.png',
            'src/assets/img/toolpage/3-draw.png', 'src/assets/img/toolpage/2-tools.png'
INFO     -  [14:06:36] Serving on http://127.0.0.1:8000/
```
* The site seems to be serving a built version of `working-with-pm.md` despite the error. `¯\_(ツ)_/¯`

## Responses to 1-3

> If you have any suggestions/critiques of specific aspects of the revised documentation content/organization, now is the time.

Thus far, I like the organization.

> But it's not really practical to merge the docs one with 'dev' itself, because then one command would have to run two servers at once [...]

Agreed. I don't think this would be a good idea.

> Other suggestions for the name of this script?

Maybe you could have two scripts (1) `docs:build` and (2) `docs:serve`? Then in the `build` script, you call `docs:build`.

> For theming/skinning, I understand the plan is to add a top bar visually identical to the one in the Numberscope tool, but below that, what's the preferred theme?

I would do this in a separate PR because I think it will require a fair amount of discussion.

## Response to 4

> (i) the package.json file looks kinda crazy

I will learn to embrace the chaos.

> (ii) you will note there are no links on the working with the package manager page, whereas there were in the corresponding section of the README file that all of that information came from. That's because there's basically no reasonable way to include parentheses in these pseudo-comments, because parens are special characters in sh which runs the script lines.

Why are there parentheses in links?

> (B) Switch this project from using the npm package manager to exclusively using the pnpm package manager.

I'm happy to do this, but I think it should be done in a separate PR unless you feel strongly that it makes sense to do it in this PR. I almost switched to Yarn V2 in another project. I'm not married to NPM.

> there is another option, which is to install the nps package (https://www.npmjs.com/package/nps) with which one can move (just) the scripts section out of package.json and into an ordinary javascript file (conventionally called "package-scripts.js").

I think having a more efficient package manager would be nice. Using Yarn V2 made me realize how slow NPM is. Also, I feel like this might not be the best long-term solution to the problem.

**Comment by liam-mulhall** [comment]
@gwhitney, TL;DR:
* looking good except for the error above
* I propose `docs:build` and `docs:serve`
* for 4, let's do PNPM

**Comment by gwhitney** [comment]
* thanks for the margin
* I will investigate the error message, there should not be any
* on item 2, i will make a docs:serve script. I am reluctant to have a separate docs:build script because the tool and docs should always be built simultaneously since they link to each other.
* on item 3, I am fine with finishing up this PR with the ReadTheDocs theme and immediately filing an issue to determine the theming.
* the markdown format for links is [TEXT](URL). Hence to have links in stuff that will be extracted to Markdown, you need parentheses.
* on item 4, thanks for adding your comments. Definitely want @katestange's input before finalizing a plan. In the meantime I will move ahead on the other points.

**Comment by liam-mulhall** [comment]
> Definitely want @katestange's input before finalizing a plan.

I hope my comments didn't give the impression that I wanted to move forward without @katestange's input. I just wanted to make my opinion on the matter clear! :)

**Comment by gwhitney** [comment]
OK, that should fix the mkdocs error. The issue was that prettier felt it could have its way with the {! ... !} extraction block in doc/working-with-pm.md, but it doesn't realize that indentation within that block matters, as the extraction specification is actually in YAML, which is whitespace-sensitive. So I restored the contents of the block and protected it with a `<!-- prettier-ignore -->` flag. A little annoying, but should be OK (and the point will be moot if we switch to pnpm, because we won't need the complicated extraction block). Will do item 2 when I get the chance, we're postponing 3 until the next PR, and we're waiting for Kate's thoughts on 4.

**Comment by gwhitney** [comment]
OK, that should take care of item 2, with the noticeable benefit that you can execute `npm run doc:serve` without having to activate the virtual environment and you should get the correct behavior. Note that I named the script "doc:serve" rather than "docs:serve" to stay consistent with the fact that we have the bulk of the documentation source in the "doc" directory rather than the "docs" directory (as some projects use). Seemed like those prefixes should be the same. Next up: incorporating documentation build into the 'build' script.

**Comment by gwhitney** [comment]
Make sure to **npm install* after this update. Now you can do `npm run build` and `npm run preview` and both of the Numberscope tool and the doc site will be there, with About and Acknowledgments replaced by links into the doc site. That just leaves the Help view. Currently on that view there are just four links:
- User Guide: this is a self-link to the Help view, so doesn't add much
- Documentation: this links to a Documentation view which just has brief descriptions of four of the five visualizers. All of this info has been transferred to the visualizer source files and is being extracted into the doc site, so I think is redundant
- Link to the github site: I propose moving this to the About page
- mailto link to the developers: I propose either eliminating this (after all, folks can file issues in github; perhaps we should provide guidance that's the preferred method of contact) or moving it to the About page.

In short, I think Help is redundant now and the Help link should be replaced with a Documentation link that goes to the top level of the doc site. Let me know your thoughts. Thanks!

**Comment by gwhitney** [comment]
OK, that moves the last bits of content as agreed and eliminates the About, Acknowledgment, Help, and UserManual views, which have all been superseded by the docsite. Since we have agreed to defer the remaining two items (summarized below for convenience) to two additional PRs to follow on the heels of this one, I feel this PR is now ready for final review and will mark it as such.

Open items for future PRs:

(3) Choice of theme and design/implementation of navbar for the docsite to dovetail with main numberscope navbar

(4) Choice of response to clumsy interpolation of documentation in package.json:

(4A) Leave as is in this PR, just deal with the clutter,
(4B) Switch to `pnpm` package manager allowing move to JSON5 format for the package file, which allows comments,
(4C) Remove scripts from package.json (as they are the only part of package.json we currently want comments in) by installing package `nps`.

**Comment by liam-mulhall** [comment]
Okay, thanks. I'll start my review once I reach good stopping point with the Cypress work. (Probably within the hour.)

**Comment by gwhitney** [comment]
> (Probably within the hour.)

Well, I mean although I'd like to discuss points (3) and (4) at today's meeting, it's not like the review has to be done by then or anything. You can do the review as it fits with your work schedule. Of course I understand if you want to get this merged for the sake of switching to Cypress, as it does reduce the number of tests you have to deal with...

**Comment by liam-mulhall** [comment]
Yes, I'd like to talk about those points in today's meeting as well.

I'll just review what I can now.

**Comment by liam-mulhall** [comment]
@gwhitney, I pulled the latest changes and it seems typecheck is failing:

```
> frontscope@0.3.0 typecheck
> vue-tsc --noEmit -p tsconfig.vitest.json --composite false

src/sequences/sequences.ts:4:46 - error TS2345: Argument of type '{ eager: boolean; }' is not assignable to parameter of type 'AssertOptions'.
  Object literal may only specify known properties, and 'eager' does not exist in type 'AssertOptions'.

4 const seqFiles = import.meta.glob('./*.ts', {eager: true})
                                               ~~~~~~~~~~~

src/sequences/sequences.ts:11:17 - error TS2352: Conversion of type '() => Promise<{ [key: string]: any; }>' to type '{ exportModule: SequenceExportModule; }' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Property 'exportModule' is missing in type '() => Promise<{ [key: string]: any; }>' but required in type '{ exportModule: SequenceExportModule; }'.

11     const mod = seqFiles[file] as {exportModule: SequenceExportModule}
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/sequences/sequences.ts:11:36
    11     const mod = seqFiles[file] as {exportModule: SequenceExportModule}
                                          ~~~~~~~~~~~~
    'exportModule' is declared here.

src/visualizers/visualizers.ts:4:46 - error TS2345: Argument of type '{ eager: boolean; }' is not assignable to parameter of type 'AssertOptions'.
  Object literal may only specify known properties, and 'eager' does not exist in type 'AssertOptions'.

4 const vizFiles = import.meta.glob('./*.ts', {eager: true})
                                               ~~~~~~~~~~~

src/visualizers/visualizers.ts:11:17 - error TS2352: Conversion of type '() => Promise<{ [key: string]: any; }>' to type '{ exportModule: VisualizerExportModule; }' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Property 'exportModule' is missing in type '() => Promise<{ [key: string]: any; }>' but required in type '{ exportModule: VisualizerExportModule; }'.

11     const mod = vizFiles[file] as {exportModule: VisualizerExportModule}
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/visualizers/visualizers.ts:11:36
    11     const mod = vizFiles[file] as {exportModule: VisualizerExportModule}
                                          ~~~~~~~~~~~~
    'exportModule' is declared here.


Found 4 errors.
```

Are you able to reproduce?

**Comment by gwhitney** [comment]
Have you done `npm install` lately? Try that, and if it still doesn't work I will investigate.

**Comment by liam-mulhall** [comment]
No, I hadn't `npm i`'d. That was the issue. Switching between branches. Thanks.

**Comment by liam-mulhall** [comment]
@gwhitney, I've looked over the file changes and I've done a fair amount of local testing. I think we are okay to merge. I just have one (minor) question:

navbar-and-home

Why is "The 'Scope" white while "About" and "Documentation" are faded? Is the plan to use the white/faded to indicate the page you are currently on? If so, I think we should add a "Home" link to the navbar because technically in the above screenshot, you're on the homepage, not the 'scope page.

Otherwise, if this is just a CSS oversight, I can fix "About" and "Documentation" so that they are white again.

**Comment by liam-mulhall** [comment]
Okay, thanks for the feedback. I'll just merge as is and tweak the styling in the mini PR I plan on doing for the UI.

**Review by liam-mulhall — APPROVED** [review]
